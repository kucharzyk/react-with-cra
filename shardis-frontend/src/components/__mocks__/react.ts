import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
