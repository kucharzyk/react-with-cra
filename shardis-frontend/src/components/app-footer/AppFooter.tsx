import * as React from 'react';
import { StatelessComponent } from 'react';
import withStyles from 'material-ui/styles/withStyles';
import { WithStyles } from 'material-ui';
import { Theme } from 'material-ui/styles';
import Paper from 'material-ui/Paper';

const styles = (theme: Theme) => ({
  root: {
    textAlign: 'center',
    padding: theme.spacing.unit
  },
});

const AppFooter: StatelessComponent<{} & WithStyles<'root'>> = (props) => (
  <Paper className={props.classes.root}>Shardis Â©2018</Paper>
);

export default withStyles(styles)<{}>(AppFooter);
