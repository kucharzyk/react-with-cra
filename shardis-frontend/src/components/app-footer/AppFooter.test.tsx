import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import AppFooter from './AppFooter';

describe('AppFooter', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AppFooter/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<AppFooter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
