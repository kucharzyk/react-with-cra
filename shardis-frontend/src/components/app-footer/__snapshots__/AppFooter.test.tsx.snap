// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter renders correctly 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 AppFooter-root-1"
>
  Shardis Â©2018
</div>
`;
